cloud/services/subnets/subnets.go

```diff
@@ -58,100 +47,100 @@ func (s *Service) getExisting(ctx context.Context, rgName string, spec *Spec) (*
 }
 
 // Reconcile gets/creates/updates a subnet.
-func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
-	subnetSpec, ok := spec.(*Spec)
-	if !ok {
-		return errors.New("Invalid Subnet Specification")
-	}
-	existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
-	if err == nil {
-		// subnet already exists, update the spec and skip creation
-		var subnet *infrav1.SubnetSpec
-		if subnetSpec.Role == infrav1.SubnetControlPlane {
-			subnet = s.Scope.ControlPlaneSubnet()
-		} else if subnetSpec.Role == infrav1.SubnetNode {
-			subnet = s.Scope.NodeSubnet()
-		} else {
+func (s *Service) Reconcile(ctx context.Context) error {
+	for i, subnetSpec := range s.Scope.SubnetSpecs() {
+		existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
+		if err == nil {
+			// subnet already exists, update the spec and skip creation
+			var subnet *infrav1.SubnetSpec
+			if subnetSpec.Role == infrav1.SubnetControlPlane {
+				subnet = s.Scope.ControlPlaneSubnet()
+			} else if subnetSpec.Role == infrav1.SubnetNode {
+				subnet = s.Scope.NodeSubnet()
+			} else {
+				return nil
+			}
+
+			subnet.Role = subnetSpec.Role
+			subnet.Name = existingSubnet.Name
+			subnet.CidrBlock = existingSubnet.CidrBlock
+			subnet.ID = existingSubnet.ID
+
+			// Continue to iterating the Subnet spec array
+			if i != len(s.Scope.SubnetSpecs())-1 {
+				continue
+			}
 			return nil
 		}
+		if !azure.ResourceNotFound(err) {
```

Comment:
Couple little nits. Please consider reformatting to the following:
```go
if err != nil && !azure.ResourceNotFound(err) {
    return errors.Wrapf(err, "failed to get subnet %q", subnetSpec.Name)
}
```

Second, please put spaces after braces to separate unrelated conditional blocks.

Reply:
done